support/controlplane-component/konnectivity-container.go

```diff
@@ -0,0 +1,187 @@
+	ResolveFromManagementClusterDNS bool
+	DisableResolver                 bool
+}
+
+func (opts *KonnectivityContainerOptions) injectKonnectivityContainer(cpContext ControlPlaneContext, podSpec *corev1.PodSpec) {
+	if opts.Mode == "" {
+		// programmer error.
+		panic("Konnectivity proxy mode must be specified!")
+	}
+
+	hcp := cpContext.HCP
+	var proxyAdditionalCAs []corev1.VolumeProjection
+	if hcp.Spec.AdditionalTrustBundle != nil {
+		proxyAdditionalCAs = append(proxyAdditionalCAs, corev1.VolumeProjection{
+			ConfigMap: &corev1.ConfigMapProjection{
+				LocalObjectReference: *hcp.Spec.AdditionalTrustBundle,
+				Items:                []corev1.KeyToPath{{Key: "ca-bundle.crt", Path: "additional-ca-bundle.pem"}},
+			},
+		})
+	}
+
+	if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.Proxy != nil && len(hcp.Spec.Configuration.Proxy.TrustedCA.Name) > 0 {
+		proxyAdditionalCAs = append(proxyAdditionalCAs, corev1.VolumeProjection{
+			ConfigMap: &corev1.ConfigMapProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: hcp.Spec.Configuration.Proxy.TrustedCA.Name,
+				},
+				Items: []corev1.KeyToPath{{Key: "ca-bundle.crt", Path: "proxy-trusted-ca.pem"}},
+			},
+		})
+	}
+
+	image := cpContext.ReleaseImageProvider.GetImage(util.CPOImageName)
+
+	podSpec.Containers = append(podSpec.Containers, opts.buildContainer(hcp, image, proxyAdditionalCAs))
+	podSpec.Volumes = append(podSpec.Volumes, opts.buildVolumes(proxyAdditionalCAs)...)
+}
+
+const certsTrustPath = "/etc/pki/tls/certs"
+
+func (opts *KonnectivityContainerOptions) buildContainer(hcp *hyperv1.HostedControlPlane, image string, proxyAdditionalCAs []corev1.VolumeProjection) corev1.Container {
+	var proxyConfig *configv1.ProxySpec
+	if hcp.Spec.Configuration != nil {
+		proxyConfig = hcp.Spec.Configuration.Proxy
+	}
+
+	command := []string{"/usr/bin/control-plane-operator"}
+	args := []string{"run"}
+	switch opts.Mode {
+	case HTTPS:
+		command = append(command, "konnectivity-https-proxy")
+		if proxyConfig != nil {
+			noProxy := proxy.DefaultNoProxy(hcp)
+			args = append(args, "--http-proxy", proxyConfig.HTTPProxy)
+			args = append(args, "--https-proxy", proxyConfig.HTTPSProxy)
+			args = append(args, "--no-proxy", noProxy)
+		}
+	case Socks5:
+		command = append(command, "konnectivity-socks5-proxy")
+		args = append(args, "--connect-directly-to-cloud-apis", strconv.FormatBool(opts.ConnectDirectlyToCloudAPIs))
+		args = append(args, "--resolve-from-guest-cluster-dns", strconv.FormatBool(opts.ResolveFromGuestClusterDNS))
+		args = append(args, "--resolve-from-management-cluster-dns", strconv.FormatBool(opts.ResolveFromManagementClusterDNS))
+		args = append(args, "--disable-resolver", strconv.FormatBool(opts.DisableResolver))
+	}
+
+	kubeconfingVolumeName := opts.KubeconfingVolumeName
+	if kubeconfingVolumeName == "" {
+		kubeconfingVolumeName = "kubeconfig"
+	}
+
+	container := corev1.Container{
+		Name:    "konnectivity-proxy",
```

Comment:
we have a use case where there are 2 konnectivity proxies in a single deployment (openshift-oauth). Is the container name something we can parameterize if needed?

Reply:
added a `DUAL` mode that will inject 2 konnectivity containers, let me know if that makes sense.